src/io/flutter/preview/PreviewView.java

```diff
@@ -285,6 +311,40 @@ public void propertyChange(PropertyChangeEvent evt) {
     contentManager.setSelectedContent(content);
   }
 
+  private ActionGroup createTreePopupActions() {
+    // The corresponding tree item has just been selected.
+    // Wait short time for receiving assists from the server.
+    for (int i = 0; i < 20 && actionToChangeMap.isEmpty(); i++) {
```

Comment:
This seems practical, but I wonder if there are any other ways of doing this. Perhaps an IntelliJ utility API?

@jwren, who I believe has used an API to see it there are any event interrupts?

Reply:
Unless there is anything asynchronous for popup menus (and I have not seen this anywhere yet), we have to wait with some timeout. There are different ways to implement this waiting, this is just one of them.

Reply:
Makes sense, I just wasn't sure whether there was some prior art in the IntelliJ APIs for helping us here. lgtm as is